sys-utils/libcmt/src/rollup.c

```diff
@@ -237,26 +236,25 @@ int cmt_rollup_read_advance_state(cmt_rollup_t *me, cmt_rollup_advance_t *advanc
     if (cmt_rollup_get_rx(me, rd)) {
         return -ENOBUFS;
     }
-    cmt_buf_t anchor[1] = {{rd->begin + 4, rd->end}};
+    cmt_buf_t frame[1];
     cmt_buf_t of[1];
 
-    size_t payload_length = 0;
-
     // clang-format off
     if (DBG(cmt_abi_check_funsel(rd, EVM_ADVANCE))
+    ||  DBG(cmt_abi_mark_frame(rd, frame))
     ||  DBG(cmt_abi_get_uint(rd, sizeof(advance->chain_id), &advance->chain_id))
-    ||  DBG(cmt_abi_get_address(rd, advance->app_contract))
-    ||  DBG(cmt_abi_get_address(rd, advance->msg_sender))
+    ||  DBG(cmt_abi_get_address(rd, &advance->app_contract))
+    ||  DBG(cmt_abi_get_address(rd, &advance->msg_sender))
     ||  DBG(cmt_abi_get_uint(rd, sizeof(advance->block_number), &advance->block_number))
     ||  DBG(cmt_abi_get_uint(rd, sizeof(advance->block_timestamp), &advance->block_timestamp))
+    ||  DBG(cmt_abi_get_uint(rd, sizeof(advance->prev_randao), advance->prev_randao.data))
```

Comment:
I am confused here. I assume when you get_uint into, say, block_number, you do big-endian to little-endian conversion. Because everybody will want to interpret that as a number natively. prev_randao is a random number, sure. But I think we should treat it more like an address and leave it alone. Otherwise people are sure to convert it to hex without flipping and end up with the reversed prev_randao. What do you think?

Reply:
agreed, fixed.